motioneye/static/js/main.js

```diff
@@ -2941,29 +2949,55 @@ function runPictureDialog(entries, pos, mediaType) {
         
         prevArrow.css('display', 'none');
         nextArrow.css('display', 'none');
+
+        /* Construct a likely mime-type with 'video/' and the file extension, then see if the 
+           browser can play it */       
+        var playable = video_container.get(0).canPlayType('video/' + entry.path.split('.').pop()) != ''
+        playButton.hide();
+        video_container.hide();
+        img.show();
+
         img.parent().append(progressImg);
         updateModalDialogPosition();
         progressImg.css('left', (img.parent().width() - progressImg.width()) / 2);
         progressImg.css('top', (img.parent().height() - progressImg.height()) / 2);
         
         img.attr('src', addAuthParams('GET', basePath + mediaType + '/' + entry.cameraId + '/preview' + entry.path));
+
+        if (playable) {
+            playButton.on('click', function() {
+                video_source.attr('src', addAuthParams('GET', basePath + mediaType + '/' + entry.cameraId + '/download' + entry.path));
+                video_container.show();
+                video_container.get(0).load();  /* Must call load() after changing <video> source */
```

Comment:
Errors must be handled. VideoJS will handle them instead of you showing nice message. Or you can just add `error` listener add display error message to user instead of silent, broken player.